src/Core/Data/TripleBuffer.hpp

```
@@ -0,0 +1,49 @@
+	*/
+    void publish() {
+        // "Release" stores ensure all previous writes to m_buffers[m_writeIndex] are visible to other threads before the index swap happens
+        m_writeIndex = m_dirtyIndex.exchange(m_writeIndex, std::memory_order_release);
+    }
+
+
+    /* Consumes the latest buffer.
+        @note This only atomically swaps the write buffer with the read buffer. To read from the read buffer, call readRef.
+		@return False if no new buffer has been published since the last consume (in which case readRef will return the most recent old buffer), otherwise True.
+    */
+    bool consume() {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)